clang/include/clang/3C/RewriteUtils.h

```diff
@@ -254,6 +254,19 @@ class RewriteConsumer : public ASTConsumer {
   void emitRootCauseDiagnostics(ASTContext &Context);
 };
 
-bool canRewrite(Rewriter &R, SourceRange &SR);
+bool canRewrite(Rewriter &R, const SourceRange &SR);
+
+
+// Rewrites the given source range with fallbacks for when the SourceRange is
+// inside a macro. This should be prefered to direct calls to ReplaceText
```

Comment:
Should we somehow use the C++ type system to limit the ability of code to call `ReplaceText`? E.g., make it a `private` method while `rewriteSourceRange` is public, or something like that?

Reply:
Closing out this topic that might have been discussed elsewhere: `ReplaceText` is a Clang API; we can't change it from public to private.  I don't think there's any other solution to this worth pursuing.  Hopefully we see that the only calls to `ReplaceText` are within `rewriteSourceRange` and realize that we should use the latter.